### DISCUSSION TITLE
How to use MAX31856 for thermocouples with tasmota?

### DISCUSSION THREAD

**Discussion by Rush56** [arendst/Tasmota#22472] [Support request]
Hi all. 

I have sucessfully used max31865 module with PT100 sensor, however RTD is pretty slow for my use.

Now i got another max31856 module with various thermocouple sensor support, and decided to go with T-type which is from my research the most accurate of all thermocouples..

I had this defined for first max31865 module and PT100 and worked fine:

> #define USE_MAX31865                           // Add support for MAX31865 RTD sensors using softSPI
> #define MAX31865_PTD_WIRES  3                  // PTDs come in several flavors, pick yours. Specific settings per sensor possible with MAX31865_PTD_WIRES1..MAX31865_PTD_WIRES6
> #define MAX31865_PTD_RES    100               // Nominal PTD resistance at 0°C (100Ω for a PT100, 1000Ω for a PT1000, YMMV!). Specific settings per sensor possible with MAX31865_PTD_RES1..MAX31865_PTD_RES6
> #define MAX31865_REF_RES    430               // Reference resistor (Usually 430Ω for a PT100, 4300Ω for a PT1000). Specific settings per sensor possible with MAX31865_REF_RES1..MAX31865_REF_RES6
> #define MAX31865_PTD_BIAS   7                  // To calibrate your not-so-good PTD. Specific settings per sensor possible with MAX31865_PTD_BIAS1..MAX31865_PTD_BIAS6

How do i go about to make this one work in tasmota with T-type thermocouple sensor? Its a blue max31865 board  with K,J,N,R,S,T,E thermocouple support.

**Comment by lalo-uy** [comment]
MAX31865 is for PT sensors
MAX31855 is for thermocuples.

**Reply by Rush56** [reply]
MAX31856 CJMCU-MAX31856
https://a.aliexpress.com/_EvKV4Nj

I've got this one. Blue board. Both are supposed to be pretty much the same...

**Reply by Rush56** [reply]
Do you think i could use MAX31855 driver to get it to work? It does however say that this driver is for K-type thermocouple. I've got a T-type...

I've noticed there is a library from adafruit for this board. Can i add that to tasmota? How do i do that if possible?
https://learn.adafruit.com/adafruit-max31856-thermocouple-amplifier?view=all

**Comment by lalo-uy** [comment]
So you got max31856 not max31865.

**Reply by Rush56** [reply]
Woops, honestly i was sure both boards were labeled the same. But yes you are right.  Its MAX31856.

So that board is not supported to use with tasmota?

**Reply by sfromis** [reply]
Not sure if they are compatible, but if so, you still have your discrepancy of K-type vs T-type.
